sunpy/physics/differential_rotation.py

```diff
@@ -159,3 +161,145 @@ def solar_rotate_coordinate(coordinate,
 
     # Return the rotated coordinates to the input coordinate frame
     return heliographic_rotated.transform_to(coordinate.frame.name)
+
+
+@u.quantity_input(dt=u.s)
+def _warp_sun_coordinates(xy, smap, dt):
+    """
+    Function that returns a new list of coordinates for each input coord.
+    This is an inverse function needed by the scikit-image `transform.warp`
+    function.
+
+    Parameters
+    ----------
+    xy :
+        Array from `transform.warp`
+    smap : `~sunpy.map`
+        Original map that we want to transform
+    dt : `~astropy.units.Quantity`
+        Desired interval to rotate the input map by solar differential rotation.
+
+    Returns
+    -------
+    xy2 : `~numpy.ndarray`
+        Array with the inverse transformation
+    """
+    # NOTE: The time is being subtracted - this is because this function
+    # calculates the inverse of the transformation.
+    rotated_time = smap.date - timedelta(seconds=dt.to(u.s).value)
+
+    # Calculate the hpc coords
+    x = np.arange(0, smap.dimensions.x.value)
+    y = np.arange(0, smap.dimensions.y.value)
+    xx, yy = np.meshgrid(x, y)
+    # the xy input array would have the following shape
+    # xy = np.dstack([xx.T.flat, yy.T.flat])[0]
```

Comment:
it's a comment talking about how to reconstruct the shape of the input array, at least that's how I read it.